lib/puppet/parser/functions/artifactory_sha1.rb

```diff
@@ -9,7 +9,7 @@ module Puppet::Parser::Functions
     raise(ArgumentError, "Invalid artifactory file info url #{args}") unless args.size == 1
 
     require 'json'
-    require 'puppet_x/bodeco/util.rb'
+    require File.dirname(__FILE__) + '/../../../puppet_x/bodeco/util.rb'
```

Comment:
Since functions are pluginsynced to the libdir, this could be loading an arbitrary version of util.rb.